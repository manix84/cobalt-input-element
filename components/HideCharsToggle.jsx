import { useEffect, useState } from "react";
import { styled } from "styled-components";
import HideCharsDarkImg from "../public/components/HideCharsToggle/hideChars.dark.png";
import HideCharsLightImg from "../public/components/HideCharsToggle/hideChars.light.png";
import ShowCharsDarkImg from "../public/components/HideCharsToggle/showChars.dark.png";
import ShowCharsLightImg from "../public/components/HideCharsToggle/showChars.light.png";

const HideCharsToggle = ({
  startAsVisible = false,
  onChange = (isVisible) => {},
  className = "",
}) => {
  const [isVisible, setIsVisible] = useState(startAsVisible);

  const handleClick = () => {
    setIsVisible((curIsVisible) => !curIsVisible);
  };

  useEffect(() => {
    onChange(isVisible);
  }, [isVisible]);

  return (
    <Container
      data-is-visible={isVisible}
      onClick={handleClick}
      className={className}
    >
      <Eye />
    </Container>
  );
};

const Container = styled.div`
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
`;

const Eye = styled.div`
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  [data-is-visible="true"] & {
    background-image: url(${HideCharsLightImg.src});
    @media (prefers-color-scheme: dark) {
      background-image: url(${HideCharsDarkImg.src});
    }
  }
  [data-is-visible="false"] & {
    background-image: url(${ShowCharsLightImg.src});
    @media (prefers-color-scheme: dark) {
      background-image: url(${ShowCharsDarkImg.src});
    }
  }
`;

export default HideCharsToggle;
