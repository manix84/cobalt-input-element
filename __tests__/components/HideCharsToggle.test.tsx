/**
 * @jest-environment jsdom
 */
import HideCharsToggle from "@/components/HideCharsToggle";
import { render } from "@testing-library/react";

describe("HideCharsToggle", () => {
  test("renders basic component unchanged", () => {
    const Toggle = render(<HideCharsToggle onChange={() => {}} />);
    expect(Toggle).toMatchSnapshot();
  });
  test("renders start visible option unchanged", () => {
    const Toggle = render(
      <HideCharsToggle startAsVisible onChange={() => {}} />
    );
    expect(Toggle).toMatchSnapshot();
  });
  test("renders start visible option unchanged", () => {
    const Toggle = render(
      <HideCharsToggle className={"className"} onChange={() => {}} />
    );
    expect(Toggle).toMatchSnapshot();
  });
});
