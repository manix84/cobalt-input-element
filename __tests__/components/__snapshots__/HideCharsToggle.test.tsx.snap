// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideCharsToggle renders basic component unchanged 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
}

.c1 {
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}

[data-is-visible="true"] .c1 {
  background-image: url(/img.jpg);
}

[data-is-visible="false"] .c1 {
  background-image: url(/img.jpg);
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="true"] .c1 {
    background-image: url(/img.jpg);
  }
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="false"] .c1 {
    background-image: url(/img.jpg);
  }
}

<body>
    <div>
      <div
        class="c0"
        data-is-visible="false"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
}

.c1 {
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}

[data-is-visible="true"] .c1 {
  background-image: url(/img.jpg);
}

[data-is-visible="false"] .c1 {
  background-image: url(/img.jpg);
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="true"] .c1 {
    background-image: url(/img.jpg);
  }
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="false"] .c1 {
    background-image: url(/img.jpg);
  }
}

<div>
    <div
      class="c0"
      data-is-visible="false"
    >
      <div
        class="c1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`HideCharsToggle renders start visible option unchanged 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
}

.c1 {
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}

[data-is-visible="true"] .c1 {
  background-image: url(/img.jpg);
}

[data-is-visible="false"] .c1 {
  background-image: url(/img.jpg);
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="true"] .c1 {
    background-image: url(/img.jpg);
  }
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="false"] .c1 {
    background-image: url(/img.jpg);
  }
}

<body>
    <div>
      <div
        class="c0"
        data-is-visible="true"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
}

.c1 {
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}

[data-is-visible="true"] .c1 {
  background-image: url(/img.jpg);
}

[data-is-visible="false"] .c1 {
  background-image: url(/img.jpg);
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="true"] .c1 {
    background-image: url(/img.jpg);
  }
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="false"] .c1 {
    background-image: url(/img.jpg);
  }
}

<div>
    <div
      class="c0"
      data-is-visible="true"
    >
      <div
        class="c1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`HideCharsToggle renders start visible option unchanged 2`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
}

.c1 {
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}

[data-is-visible="true"] .c1 {
  background-image: url(/img.jpg);
}

[data-is-visible="false"] .c1 {
  background-image: url(/img.jpg);
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="true"] .c1 {
    background-image: url(/img.jpg);
  }
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="false"] .c1 {
    background-image: url(/img.jpg);
  }
}

<body>
    <div>
      <div
        class="c0 className"
        data-is-visible="false"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  cursor: pointer;
  outline: 0 none;
  border: 1px solid currentColor;
  border-radius: 4px;
}

.c1 {
  display: block;
  height: 25px;
  width: 25px;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}

[data-is-visible="true"] .c1 {
  background-image: url(/img.jpg);
}

[data-is-visible="false"] .c1 {
  background-image: url(/img.jpg);
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="true"] .c1 {
    background-image: url(/img.jpg);
  }
}

@media (prefers-color-scheme: dark) {
  [data-is-visible="false"] .c1 {
    background-image: url(/img.jpg);
  }
}

<div>
    <div
      class="c0 className"
      data-is-visible="false"
    >
      <div
        class="c1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
